SingularityService/src/main/java/com/hubspot/singularity/SingularityLeaderController.java

```diff
@@ -71,6 +85,40 @@ public void start() {
 
   public void stop() {
     statePoller.finish();
+    TIMER.cancel();
+  }
+
+  @Override
+  public void stateChanged(CuratorFramework client, ConnectionState newState) {
+    stateHandlerLock.lock();
+    try {
+      scheduler.setZkConnectionState(newState);
+      ConnectionState previous = lastState.get();
+      if (!newState.isConnected()) {
+        scheduler.notLeader();
```

Comment:
Logging here that connection has been lost and we're going to hang out for X millis would be useful for debugging